src/CMakeLists.txt

```diff
@@ -23,6 +23,7 @@ foreach(libtype shared static)
       $<TARGET_OBJECTS:hiopLinAlg>
       $<TARGET_OBJECTS:hiopOptimization>
       $<TARGET_OBJECTS:hiopUtils>
+      $<TARGET_OBJECTS:hiopInterface>
```

Comment:
You should only have to add the `hiopInterface` object once in this `add_library` command. Since the object is build regardless of shared/static libs, I think you should just be able to remove `$<$<BOOL:${HIOP_BUILD_SHARED}>:$<TARGET_OBJECTS:hiopInterface>>` entirely @nychiang